### PR TITLE
9 create query to fetch a list of users

### PR DESCRIPTION
**Pull request by AlbinNils98**
## Overview:
This pull request introduces a GraphQL query implementation to fetch all user data. It implements the users query in the GraphQL schema and connects it to the underlying user data through a service layer in the application. The users' details are fetched and returned as a set of UserDto objects.

### 1. GraphQL Schema (Schema Definition):

- Query Type: A new query users is added to the schema to fetch a list of users.

### 2. UserController:

- The controller exposes the GraphQL query for fetching users.
- The users() method is mapped to the users GraphQL query, using the @QueryMapping annotation.
- It delegates the logic to the UserService for fetching the data.

### 3. UserService:

The service handles the core logic of fetching all users from the database and converting them into UserDto objects.
The getAllUsers() method uses the repository to get all user entities and maps them to UserDto objects.

### 4. UserDto:

- The UserDto record represents a simplified data transfer object (DTO) for user information.
- The from() method is used to convert the User entity into a UserDto object.

## Changes Made:
- GraphQL Schema: Defined a users query to return a list of User objects with name and email fields.
- UserController: Created a controller to handle the users query and map it to a service method.
- UserService: Implemented the logic for fetching all users from the repository and converting them to DTOs.
- UserDto: Added a DTO class to represent user data in a simplified form.
- Graphiql: Added graphiql in application.properties for easier testing of the queries.
- Sql: Changed sql schema for user so the user has a first- and lastname

## Good to know:
To test the graphql queries, for development purposes, you need to disable csrf in the developmentSecurityFilterchain as the graphql requests are made with the POST-method.

## Summary by CodeRabbit

- **New Features**
	- Introduced a GraphQL interface for querying user data.
	- Added a controller for managing user-related operations.
	- Implemented a service for retrieving user information.
	- Created a data transfer object for user representation.

- **Bug Fixes**
	- Enhanced database schema to accommodate separate first and last name fields.

- **Documentation**
	- Updated GraphQL schema to reflect new user data structure.

- **Chores**
	- Modified SQL scripts for database population to align with the new user structure.

### CODE REVIEW

File: src/main/java/org/fungover/system2024/user/UserController.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  private final UserService userService;
+
+  public UserController(UserService userService) {
+    this.userService = userService;
+  }
+
+  @QueryMapping
+  Iterable<UserDto> users(){
+    return userService.getAllUsers();
+  }
```
[the remark this answers is not in the document]
Comment (AlbinNils98) [reply src/main/java/org/fungover/system2024/user/UserController.java:21]: Will fix in issue #65

**Comment by MikaelaMathisson** [comment]
@AlbinNils98 Theres conflicts that needs to be resolved :)

**Comment by AlbinNils98** [comment]
> @AlbinNils98 Theres conflicts that needs to be resolved :)

Fixed!

**Review by MasakiShiraishi — APPROVED** [review]
It looks great! This pull request is well-structured and clearly outlines the implementation of the users GraphQL query.

**Pull request merged**